Pip package

michael-erasmus opened the pull request:
Hi @davidgasquez 

As we discussed earlier, I would love to share the PR I made for moving the project to a installable package. As I mentioned before, I can still see extracting some of the more generic Redshift data loading code into a separate package as well, but I haven't done that here yet :D

No rush on this one, but I would really appreciate your thoughts/ideas!

michael-erasmus committed:
Move to a python module and add a run command to the crawler

The run command will query the database to find the last date of events
in the subscription events table. It will then backfill from that date
up until now, after which it will keep polling stripe for new events
every minute and load them

michael-erasmus committed:
Convert project to a standalone package

michael-erasmus commented on Pipfile:
```diff
@@ -6,15 +6,7 @@ verify_ssl = true
-decorator = "*"
-click = "*"
-retrying = "*"
+"e1839a8" = {path = ".", editable = true}
```
Moving all the requirements to the `setup.py` file, pipenv will need to be aware of the package itself.

michael-erasmus commented on bin/stripe-pipeline:
```diff
@@ -0,0 +1,4 @@
+#!/usr/bin/env python
+
+from stripe_pipeline.cli import cli
```
A tiny wrapper executable for the `cli` module

davidgasquez replied:
I think this file could be deleted if the setup script uses the [`entrypoint` parameter](http://click.pocoo.org/5/setuptools/#introduction). The entrypoint specifies the name of the binary package and which function should it reference. 

In this case could be something like:

```python
entry_points={
    'console_scripts': [
        'stripe-pipeline=stripe_pipeline.cli.cli'
    ]
}
```

michael-erasmus replied:
Whoa, didn't know about that! I'll try that out!

michael-erasmus commented on setup.py:
```diff
@@ -0,0 +1,21 @@
+          'click',
+          'retrying'
+      ],
+      scripts=['bin/stripe-pipeline'])
```
This refers to the `stripe-pipeline` executable wrapper

michael-erasmus commented on stripe_pipeline/cli.py:
```diff
@@ -0,0 +1,15 @@
+    pass
+
+
+cli.add_command(crawler)
```
Write now the cli just has a single `crawler` command, but I plan on adding more commands, for things like initializing the database and possible add-hoc tasks

michael-erasmus replied:
Thanks @davidgasquez! That worked great, added it in [this commit](https://github.com/bufferapp/stripe-pipeline/pull/1/commits/c97db2db982acf4a64169b45b4903aa64db68323)

davidgasquez reviewed (approved):
Hey there @michael-erasmus, nice refactoring/packaging! 😄 

I've left one comment but LGTM overall!

michael-erasmus committed:
No need for stripe_pipeline wrapper script

Thanks to [this excellent suggestion](https://github.com/bufferapp/stripe-pipeline/pull/1#discussion_r164685360)
